paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/cvcrecollection/CvcRecollectionInteractor.kt

```diff
@@ -0,0 +1,47 @@
+package com.stripe.android.paymentsheet.paymentdatacollection.cvcrecollection
+
+import com.stripe.android.ui.core.elements.CvcController
+import com.stripe.android.ui.core.elements.CvcElement
+import com.stripe.android.uicore.elements.IdentifierSpec
+import com.stripe.android.uicore.utils.mapAsStateFlow
+import com.stripe.android.uicore.utils.stateFlowOf
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+
+internal interface CvcRecollectionInteractor {
+    val viewState: CvcRecollectionViewState
+    val cvcCompletionState: StateFlow<CvcCompletionState>
+}
+
+internal class DefaultCvcRecollectionInteractor(
+    args: Args,
+    private val scope: CoroutineScope
+) : CvcRecollectionInteractor {
+    private val controller = CvcController(cardBrandFlow = stateFlowOf(args.cardBrand))
+    private val element = CvcElement(
+        IdentifierSpec(),
+        controller
+    )
+    override val viewState = CvcRecollectionViewState(
+        cardBrand = args.cardBrand,
+        lastFour = args.lastFour,
+        cvc = null,
+        isTestMode = args.isTestMode,
+        element = element
+    )
+
+    override val cvcCompletionState: StateFlow<CvcCompletionState>
+        get() = controller.isComplete.mapAsStateFlow { isComplete ->
```

Comment:
to simplify this, can you define this without the `get()`? I think that might element the need for the `stateIn`, etc

Reply:
ahh I gotcha, I think you can also use this by creating the scope with a dispatcher and then using a test dispatcher: https://github.com/stripe/stripe-android/blob/0f99d2e75cdaa3137c7636175a90047da070ade4/paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenInteractor.kt#L65

This would let you define the stateflow as normal